posthog/hogql_queries/events_query_runner.py

```diff
@@ -217,7 +218,35 @@ def to_query(self) -> ast.SelectQuery:
                     return events_query
 
                 # Choose table based on useRecentEventsTable flag
-                table_name = "recent_events" if self.query.useRecentEventsTable else "events"
+                # Only use recent_events if the date range is within the last 7 days
+                use_recent_events = self.query.useRecentEventsTable
+                if use_recent_events:
+                    # Check if the query date range falls within the last 7 days
+                    current_time = now()
```

Comment:
Don't we have some now provider for testing purposes we should use?

Reply:
I have no idea? do you know what it might be?